src/server/readpool/metrics.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use prometheus::*;
+
+lazy_static! {
+    /// Command durations. `cmd_type` can be `kv` or `raw`. `stage` can be `snapshot` or `process`.
+    /// Failed stages are still counted. Notice that subsequent stages may not exist. For example,
+    /// if `snapshot` fails, duration of a failed `snapshot` is counted, but there will be no
+    /// `process` stages.
+    pub static ref COMMAND_HISTOGRAM_VEC: HistogramVec =
```

Comment:
These metrics seem temporary if you would like to share metrics with coprocessor later, maybe you could do it in this PR?

Reply:
Could you explain "temporary" for me? Storage and coprocessor indeed will share common metrics later (also coprocessor can have some other metrics).